controllers/storagecluster/reconcile.go

```diff
@@ -167,6 +167,11 @@ func (r *StorageClusterReconciler) Reconcile(ctx context.Context, request reconc
 	// Reconcile changes to the cluster
 	result, reconcileError := r.reconcilePhases(sc, request)
 
+	// call detectPlatform func to detect the platform of running cluster
+	if err := r.platform.detectPlatform(r.Client); err != nil {
```

Comment:
Running this every Reconcile() still seems a bit much... I'd like to figure out why we can't do it when we set up the reconciler in the first place, but I'll take it for now.

Something else to consider: Do we really want to continue reconciliation if this returns an error? Not being able to detect the platform could be a temporary network error, and thus if it resolves we would get a different value in a later reconciliation and end up altering all related configuration. This could be a problem. I would say we should `return` here. If `ErrNotFound` is a possibility, we should account for that in `detectPlatform()` and not return an error.

Reply:
How about I add if condition in detectPlatform() func to check if get platform gives me a platform value that way I can skip the error part in the func. I think it resolves both the problems. Thoughts?

Reply:
@jarrpa  any thoughts on how could we handle error from `detectPlatform()` ? What if we don't return error in `Reconcile()` and then the `platform` value is not updated even when the network issue resolves?